Babylon/commands/api/workspaces/create.py

```diff
@@ -1,91 +1,89 @@
 import pathlib
 
 from logging import getLogger
-from posixpath import basename
-import sys
 from typing import Any, Optional
-from click import Context, argument, pass_context
 from click import command
 from click import option
 from click import Path
+
+from Babylon.commands.api.workspaces.service.api import WorkspaceService
+
 from Babylon.utils.checkers import check_ascii, check_email
-from Babylon.utils.decorators import inject_context_with_resource, wrapcontext
+from Babylon.utils.decorators import (
+    wrapcontext,
+    retrieve_state,
+)
 from Babylon.utils.decorators import timing_decorator
-from Babylon.utils.messages import SUCCESS_CONFIG_UPDATED, SUCCESS_UPDATED
-from Babylon.utils.typing import QueryType
 from Babylon.utils.response import CommandResponse
 from Babylon.utils.decorators import output_to_file
 from Babylon.utils.environment import Environment
 from Babylon.utils.credentials import pass_azure_token
-from Babylon.utils.request import oauth_request
 
 logger = getLogger("Babylon")
 env = Environment()
 
 
 @command()
 @wrapcontext()
-@pass_context
 @timing_decorator
 @output_to_file
 @pass_azure_token("csm_api")
-@option("--file",
-        "w_file",
-        type=Path(path_type=pathlib.Path),
-        help="Your custom workspace description file (yaml or json)")
+@retrieve_state
+@option(
+    "--payload",
+    "payload",
+    type=Path(path_type=pathlib.Path),
+    help="Your custom workspace description file (yaml or json)",
+)
 @option("--email", "security_id", help="Workspace security_id aka email")
-@option("--role", "security_role", required=True, default="admin", help="Workspace security role")
-@option("--select", "select", is_flag=True, default=True, help="Save this new workspace in configuration")
-@argument("name", type=QueryType())
-@inject_context_with_resource({
-    "azure": ['email'],
-    "api": ['url', 'organization_id', 'workspace_key', 'solution_id', 'run_templates'],
-    "powerbi": ['dashboard_view', 'scenario_view']
-})
-def create(ctx: Context,
-           context: Any,
-           azure_token: str,
-           name: str,
-           w_file: Optional[pathlib.Path] = None,
-           security_id: Optional[str] = None,
-           security_role: Optional[str] = None,
-           select: bool = False) -> CommandResponse:
+@option(
+    "--role",
+    "security_role",
+    required=True,
+    default="admin",
+    help="Workspace security role",
+)
+@option("--organization-id", type=str)
+@option("--name", "name", type=str)
+def create(
+    state: Any,
+    azure_token: str,
+    organization_id: str,
+    name: Optional[str],
+    payload: Optional[pathlib.Path] = None,
```

Comment:
shouldn't payload be required when creating a workspace?

Reply:
excellent question, I will find out :laughing: